File: javascript/grade-school/grade-school.js
function School() {
    this.db = {};
};

School.prototype.roster = function() {
    return this.db;
};

School.prototype.add = function(name, grade) {
    names = this.db[grade] || [];
    names.push(name);
    this.db[grade] = names.sort();
};

School.prototype.grade = function(grade) {
    return this.db[grade] || [];
}

module.exports = School;

File: cpp/grade-school/grade_school.cpp
#include "grade_school.h"
#include <algorithm>

namespace grade_school
{
    const grade_container school::roster()
    {
        return _db;
    }

    void school::add(const string& name, const int& grade)
    {
        _db[grade].emplace_back(name);
        sort(_db[grade].begin(), _db[grade].end());
    }

    const vector<string> school::grade(const int& grade)
    {
        return _db[grade];
    }
}